dao-utils/src/test_env.rs

```diff
@@ -332,13 +332,13 @@ fn blake2b<T: AsRef<[u8]>>(data: T) -> [u8; 32] {
 }
 
 fn parse_error(err: engine_state::Error) -> Error {
-    if let engine_state::Error::Exec(exec_err) = err {
+    if let engine_state::Error::Exec(exec_err) = &err {
```

Comment:
Maybe that way? We don't need any `&` and `*`, by using `match` we can simply handle next specific errors.
```rust
fn parse_error(err: engine_state::Error) -> Error {
    if let engine_state::Error::Exec(exec_err) = err {
        match exec_err {
            ExecutionError::Revert(ApiError::User(id)) => Error::from(id),
            ExecutionError::InvalidContext => Error::InvalidContext,
            _ => panic!("{}", exec_err.to_string()),
        }
    } else {
        panic!("{}", err.to_string())
    }
}
```